lib/mix/lib/mix/tasks/deps.clean.ex

```diff
@@ -112,7 +133,8 @@ defmodule Mix.Tasks.Deps.Clean do
       |> Path.join(to_string(app))
       |> Path.wildcard()
       |> maybe_warn_for_invalid_path(app)
-      |> Enum.each(&File.rm_rf!/1)
+      |> Enum.map(&File.rm_rf/1)
+      |> maybe_warn_failed_file_deletion(app)
```

Comment:
```suggestion
      |> Enum.each(fn x -> x |> File.rm_rf() |> maybe_warn_failed_file_deletion(app) end)
```

And then we no longer need to deal with the list case? :)